.github/workflows/test.yaml

```diff
@@ -661,10 +656,26 @@ jobs:
           echo "Collected labels: $LABELS"
       shell: bash
 
+    - name: Get changed files
+      id: get-changed-files
+      uses: tj-actions/changed-files@v34.5.1
+
+    - name: Detect changes files
+      id: detect-changes
+      run: |
+          CHANGED_FILES="${{ steps.get-changed-files.outputs.all_changed_files }}"
+          for file in CHANGED_FILES; do
+              if [[ "$file" == tests/nested/**/* ]] || [[ "$file" == tests/lib/nested.sh ]]; then
+                  echo "changes_found=true" >> $GITHUB_OUTPUT
```

Comment:
nitpick: maybe the variable should mention nested somewhere (`nested_tests_changed`?)